R/json_pack_join.R

```diff
@@ -0,0 +1,36 @@
+#' JSON pack join
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' A wrapper around [pack_join()] which stores the joined data into a JSON column.
+#' `json_pack_join()` returns all rows and columns in `x` with a new JSON columns that contains all packed matches from `y`.
+#'
+#' @inheritParams dplyr::nest_join
+#' @param x,y A pair of data frames or data frame extensions (e.g. a tibble).
+#' @seealso [pack_join], [json_nest_join]
+#' @export
+#' @examples
+#' df1 <- tibble::tibble(x = 1:3)
+#' df2 <- tibble::tibble(x = c(1, 1, 2), y = c("first", "second", "third"))
+#' df3 <- json_pack_join(df1, df2)
+#' df3
+#' df3$df2
+json_pack_join <- function(x, y, by = NULL, ..., copy = FALSE, keep = FALSE, name = NULL) {
+  UseMethod("json_pack_join")
+}
+
+#' @export
+json_pack_join.data.frame <- function(x, y, by = NULL, ..., copy = FALSE, keep = FALSE, name = NULL) {
+  check_suggested("jsonlite", TRUE)
```

Comment:
What does `TRUE` mean here? Logical literals passed as arguments should always be named.

Fixing later.